File: pyworkspace/packages/a2a-sdk/src/trogonai_a2a_sdk/subject.py
from .types import AgentId


CALLER_JWT_HEADER = "A2a-Caller-Jwt"
DEFAULT_PURPOSE = "default"
MAX_ACT_CHAIN_DEPTH = 8


def agent_request_subject(agent: AgentId) -> str:
    return f"mcp.agent.{agent.tenant}.{agent.name}.request"


def agent_queue_group(agent: AgentId) -> str:
    return f"agent-{agent.tenant}-{agent.name}"

File: tsworkspace/packages/a2a-sdk/src/subject.ts
import type { AgentId } from "./types.js";

export function agentRequestSubject(agent: AgentId): string {
  return `mcp.agent.${agent.tenant}.${agent.name}.request`;
}

export function agentQueueGroup(agent: AgentId): string {
  return `agent-${agent.tenant}-${agent.name}`;
}

export const CALLER_JWT_HEADER = "A2a-Caller-Jwt";
export const DEFAULT_PURPOSE = "default";
export const MAX_ACT_CHAIN_DEPTH = 8;
